planner/core/find_best_task.go

```diff
@@ -415,12 +416,112 @@ func (ds *DataSource) tryToGetDualTask() (task, error) {
 	return nil, nil
 }
 
+type columnWithLen struct {
```

Comment:
We can move it to `planner/util`.
And add type alias `type ColumnsWithLen []*ColumnWithLen`
Then move the functions `firstDominateSecond`, `compareColumnsWithLens` to `ColumnsWithLen`'s member function.